### DISCUSSION TITLE
Vue - Control pinia store with storybook controls

### DISCUSSION THREAD

**Discussion by daaanny90** [storybookjs/storybook#26408] [Help]
### Summary

Hey there,
I am using Storybook on a Vue3 project with Pinia as a store manager.

I am using a pattern where each component gets data directly from the store without receiving props.

I really would like to use the Storybook controls to manipulate the store in order to see the changes in my component.

So, for example, I have a component like this:

```vue
// myComponent.vue
<script>
 import {useMyStore} from '@/myStore';

const store = useMyStore();
</script>

<template>
  <h1>{{ store.myTitle }}</h1>
</template>
```

```javascript
// myStore.js

const useMyStore = defineStore('my-store', {
  state: {
    myTitle: 'hello'
  }
})
```

```javascript
// myComponent.stories.js
import MyComponent from '@/myComponent';

const meta = {
  ...
  argTypes: {
    myTitle: {
      store: useMyStore(), // this would be nice to have
      control: { type: "text" },
    },
  },
}
```

And then a control `myTitle` would appear, and every time I update the value of the control, the corresponding value in the Pinia store will be updated, and so will be also my component.

Am I the only one with this need? It looks weird, I think it's a quite common use case and many would benefit from it.

That's why I thought that I am the one that is doing something wrong or I just do not know how to do it in Storybook.

Any ideas?

**Comment by cbebry** [comment]
I have a use case for this as well. I've noticed that older versions of Storybook (via various docs/articles online) seem to have touched on using the created hook for mocking pinia values, but this does not seem to be as functional or recommended in the current stable version. I think there are real valid use cases out there for being able to clearly supply values in the pinia store for Storybook stories for the sake of mocking scenarios, without needing to refactor the components to be strictly presentational.
